gossipd/gossip_generation.c

```diff
@@ -287,6 +287,11 @@ static bool update_own_node_announcement(struct daemon *daemon,
 					 &only_missing_tlv)) {
 			if (always_refresh)
 				goto send;
+			/* Update if old announcement is at least 7 days old. */
+			if (timestamp > self->bcast.timestamp &&
+			    timestamp - self->bcast.timestamp >
+			    GOSSIP_PRUNE_INTERVAL(daemon->rstate->dev_fast_gossip_prune) / 2)
+				goto send;
```

Comment:
setup_force_nannounce_regen_timer runs every 24 hours and calls this with always_refresh=true, so I don't see how this helps?

Reply:
It should, but setup_force_nannounce_regen_timer is not being called in the first place.  It slips through on this return without ever initiating the timer:
```
	if (self && self->bcast.index) {
		u32 next;
		bool only_missing_tlv;

		if (!nannounce_different(daemon->rstate->gs, self, nannounce,
					 &only_missing_tlv)) {
			if (always_refresh)
				goto send;
			return false;
		}
```
Actually, it frees the timer prior to this point, so I'm not sure even that would cover it.

Reply:
Then I think the right answer is simply to call setup_force_nannounce_regen_timer the first time?  A bit tricky, since we're returning true on that path, but we can remove that since nobody actually uses the return value!

```diff
diff --git a/gossipd/gossip_generation.c b/gossipd/gossip_generation.c
index 4c70b1e5c..e2c38296b 100644
--- a/gossipd/gossip_generation.c
+++ b/gossipd/gossip_generation.c
@@ -251,8 +251,7 @@ static void setup_force_nannounce_regen_timer(struct daemon *daemon);
  * the routing.c code like any other `node_announcement`.  Such announcements
  * are only accepted if there is an announced channel associated with that node
  * (to prevent spam), so we only call this once we've announced a channel. */
-/* Returns true if this sent one, or has arranged to send one in future. */
-static bool update_own_node_announcement(struct daemon *daemon,
+static void update_own_node_announcement(struct daemon *daemon,
 					 bool startup,
 					 bool always_refresh)
 {
@@ -287,7 +286,10 @@ static bool update_own_node_announcement(struct daemon *daemon,
 					 &only_missing_tlv)) {
 			if (always_refresh)
 				goto send;
-			return false;
+			/* First time?  Start regen timer. */
+			if (!daemon->node_announce_regen_timer)
+				goto reset_timer;
+			return;
 		}
 
 		/* Missing liquidity_ad, maybe we'll get plugin callback */
@@ -302,7 +304,7 @@ static bool update_own_node_announcement(struct daemon *daemon,
 					       time_from_sec(delay),
 					       update_own_node_announcement_after_startup,
 					       daemon);
-			return true;
+			return;
 		}
 		/* BOLT #7:
 		 *
@@ -324,7 +326,7 @@ static bool update_own_node_announcement(struct daemon *daemon,
 					       time_from_sec(next - timestamp),
 					       update_own_node_announcement_after_startup,
 					       daemon);
-			return true;
+			return;
 		}
 	}
 
@@ -334,8 +336,6 @@ send:
 reset_timer:
 	/* Generate another one in 24 hours. */
 	setup_force_nannounce_regen_timer(daemon);
-
-	return true;
 }
 
 static void update_own_node_announcement_after_startup(struct daemon *daemon)
```